plugins/libplugin.c

```diff
@@ -582,6 +599,31 @@ static void setup_command_usage(const struct plugin_command *commands,
 	}
 }
 
+static void call_plugin_timer(struct plugin_conn *rpc, struct timer *timer)
+{
+	struct plugin_timer *t = container_of(timer, struct plugin_timer, timer);
+
+	in_timer = true;
+	t->cb();
+	tal_free(t);
```

Comment:
Instead of `tal_free` here, could it be possible to just re-parent the `t` to some `tmpctx` that is then freed?

See `trigger_invoice_waiter_resolve` in `wallet/invoices.c` for example of this technique.

That way it is OK for a timer callback to call code that destroys the  `struct plugin_timer` (which may be better if it unifies code that has to do cleanup in both "normal" and "timeout" paths of some command).

     tal_t *tmpctx = tal(NULL, int);
     struct plugin_timer *t = container_of(timer, struct plugin_timer, timer);
     tal_steal(tmpctx, t);
     in_timer = true;
     t->cb();
     tal_free(tmpctx);

Reply:
Yes, that's my usual technique where the caller may free it.  But there's no reason for the caller to free it, at least in the current examples.

Reply:
Still suggest to keep it so caller is free to free it.  Rather than have a counterexample in the future.